server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -1068,14 +1074,20 @@ public void testScrollCreatedOnReplica() throws Exception {
         List<String> currentFiles = List.of(replicaShard.store().directory().listAll());
         assertTrue("Files should be preserved", currentFiles.containsAll(snapshottedSegments));
 
-        client(replica).prepareClearScroll().addScrollId(searchResponse.getScrollId()).get();
+        ClearScrollResponse clearResponse = client().prepareClearScroll().addScrollId(searchResponse.getScrollId()).get();
+        assertThat(clearResponse.isSucceeded(), is(true));
+        assertThat(clearResponse.getNumFreed(), greaterThan(0));
+        assertThat(clearResponse.status(), equalTo(RestStatus.OK));
 
-        assertBusy(
-            () -> assertFalse(
+        assertBusy(() -> {
```

Comment:
For my own edification, what exactly is the asynchronous process that we're waiting for here?

Reply:
Good question...My understanding here was segments are deleted async after the scroll context is cleared, but tracing through i don't see that happening asynchronously...  I traced this through again from the clear request and it should issue a delete synchronously with the scroll clear.  SearchContext is closed -> searchSupplier is closed -> underlying reader is closed -> fileDeleter decref is invoked with the ref'd files of the infos for both primary and replicas. Will close this pr and revisit this.  Either that delete is not occurring on the scroll clear as I expect or these segments are ref'd elsewhere.

Reply:
Closing this in favor of https://github.com/opensearch-project/OpenSearch/pull/12077.  There is no async behavior here, the files should be purged with the scroll clear.